src/Main.c

```diff
@@ -131,61 +117,50 @@ int comparePerformance(){
 }
 
 // for sort function that take array and array length as parameters
-void outputPerformance(char name[], int* array_of_inputs[], int sizes[], int length,  int (*sort_function)(int[], int)){
-  clock_t start_t, end_t, total_t, total_t_ascending, total_t_descending;
+void outputPerformance(char name[], int* array_of_inputs[], int sizes[], int length) {
+  // TODO: make useable unit for ticks
+  clock_t total_t, total_t_ascending, total_t_descending;
   printf("----------%s------------\n", name);
   printf("elements\tduration:\trandom\tascending\tdescending\n" );
   for (int i = 0; i<length; i++){
-  start_t = clock();
-  sort_function(array_of_inputs[i],sizes[i]);
-  end_t = clock();
-  total_t = (double)(end_t - start_t);
+    // sorted random
+    total_t = measureSort(name, array_of_inputs[i], sizes[i]);
 
-  // time if already ascending sorted array
-  start_t = clock();
-  sort_function(array_of_inputs[i],sizes[i]);
-  end_t = clock();
-  total_t_ascending = (double)(end_t - start_t);
+    // already sorted ascending
+    total_t_ascending = measureSort(name, array_of_inputs[i], sizes[i]);
 
-  bubbleSortDescending(array_of_inputs[i], sizes[i]);
+    // make sort descending
+    bubbleSortDescending(array_of_inputs[i], sizes[i]);
 
-   // time if already descending sorted array
-  start_t = clock();
-  sort_function(array_of_inputs[i],sizes[i]);
-  end_t = clock();
-  total_t_descending = (double)(end_t - start_t);
+    total_t_descending = measureSort(name, array_of_inputs[i], sizes[i]);
 
-  printf("%d\t\t\t\t%lu\t%lu\t\t%lu\n", sizes[i], total_t, total_t_ascending, total_t_descending);
+    printf("%d\t\t\t\t%lu\t%lu\t\t%lu\n", sizes[i], total_t, total_t_ascending, total_t_descending);
   }
 }
 
-// for sort function that take array, start index and end index as parameter
-void outputPerformance2(char name[], int* array_of_inputs[], int sizes[], int length, int (*sort_function)(int[], int, int)){
-  clock_t start_t, end_t, total_t, start_t_ascending, end_t_ascending, total_t_ascending, start_t_descending, end_t_descending, total_t_descending;
-  printf("----------%s------------\n", name);
-  printf("elements\tduration:\trandom\tascending\tdescending\n" );
-  for (int i = 0; i<length; i++){
+clock_t measureSort(char name[], int array_of_inputs[], int length) {
```

Comment:
measure Sort possible as helper function in Helper.c and import via header?

Reply:
I moved it into Sort.c (fits better I think) and also moved executeSort (as it is a dependency of measureSort) and the char arrays too, because the functions rely now on them (after using the variables for the string comparison)